src/server/scripts/Northrend/Naxxramas/boss_anubrekhan.cpp

```diff
@@ -37,141 +37,132 @@ enum Spells
-                cr->AI()->Talk(EMOTE_SCARAB);
-            }
-        }
+    void KilledUnit(Unit* victim) override
+    {
+        if (!victim->IsPlayer())
+            return;
+
+        Talk(SAY_SLAY);
+        victim->CastSpell(victim, SPELL_SUMMON_CORPSE_SCARABS_5, true, nullptr, nullptr, me->GetGUID());
+        instance->StorePersistentData(PERSISTENT_DATA_IMMORTAL_FAIL, 1);
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        me->CallForHelp(30.0f);
+        Talk(SAY_AGGRO);
 
-        void JustDied(Unit*  killer) override
+        if (!summons.HasEntry(NPC_CRYPT_GUARD))
+            SummonCryptGuards();
+        if (!Is25ManRaid())
         {
-            BossAI::JustDied(killer);
-            instance->DoStartTimedAchievement(ACHIEVEMENT_TIMED_TYPE_EVENT, ACHIEV_TIMED_START_EVENT);
+            ScheduleUniqueTimedEvent(17500ms, [&] {
+                me->SummonCreature(NPC_CRYPT_GUARD, cryptguardPositions[2], TEMPSUMMON_CORPSE_TIMED_DESPAWN, 60000);
+            }, EVENT_SPAWN_CRYPT_GUARDS_1);
         }
 
-        void KilledUnit(Unit* victim) override
-        {
-            if (!victim->IsPlayer())
-                return;
+        ScheduleTimedEvent(15s, [&] {
+            DoCastRandomTarget(SPELL_IMPALE);
+        }, 20s);
 
-            Talk(SAY_SLAY);
-            victim->CastSpell(victim, SPELL_SUMMON_CORPSE_SCRABS_5, true, nullptr, nullptr, me->GetGUID());
-            instance->StorePersistentData(PERSISTENT_DATA_IMMORTAL_FAIL, 1);
-        }
+        ScheduleTimedEvent(70s, 2min, [&] {
+            Talk(EMOTE_LOCUST);
+            DoCastSelf(SPELL_LOCUST_SWARM);
+            uint32 _cryptGuardEventCount = EVENT_SPAWN_CRYPT_GUARDS_EXTRA + _extraCryptGuardCount;
 
-        void JustEngagedWith(Unit* who) override
-        {
-            BossAI::JustEngagedWith(who);
-            me->CallForHelp(30.0f);
-            Talk(SAY_AGGRO);
-
-            if (!summons.HasEntry(NPC_CRYPT_GUARD))
-                SummonCryptGuards();
-            if (!Is25ManRaid())
-            {
-                me->m_Events.AddEventAtOffset([&]
-                {
-                    me->SummonCreature(NPC_CRYPT_GUARD, 3331.217f, -3476.607f, 287.074f, 3.269f, TEMPSUMMON_CORPSE_TIMED_DESPAWN, 60000);
-                }, Milliseconds(urand(15000, 20000)));
-            }
-
-            ScheduleTimedEvent(15s, [&] {
-                DoCastRandomTarget(SPELL_IMPALE);
-            }, 20s);
-
-            ScheduleTimedEvent(70s, 2min, [&] {
-                Talk(EMOTE_LOCUST);
-                DoCastSelf(SPELL_LOCUST_SWARM);
-
-                me->m_Events.AddEventAtOffset([&]
-                {
-                    me->SummonCreature(NPC_CRYPT_GUARD, 3331.217f, -3476.607f, 287.074f, 3.269f, TEMPSUMMON_CORPSE_TIMED_DESPAWN, 60000);
-                }, 3s);
-
-            }, 90s);
-
-            me->m_Events.AddEventAtOffset([&]
```

Comment:
This was set that way because the taskscheduler will skip seconds when the creature is casting. Another alternative to this is the ScheduleEnrageTimer() which also does the same